### DISCUSSION TITLE
Disable subsampling for HEIF

### DISCUSSION THREAD

**Discussion by Moarc** [ImageMagick/ImageMagick#5190] [Help]
`heif-enc -L -p chroma=444 --matrix_coefficients=0 file.png` produces a HEIF file with 4:4:4 sampling, while `magick convert -quality 100 -sampling-factor 4:4:4 -define heic:chroma=444 file.tif file.heic` gives me 4:2:0. What else do I have to specify to disable subsampling? I tried various formats (1x1,1x1,1x1, 4:4:4, etc.), to no avail.

**Comment by urban-warrior** [comment]
If compression quality is 100, ImageMagick sets the chroma to 4:4:4.  However, it is overridden for a sRGB compatible colorspace.  Try `-colorspace YCbCr -quality 100' to ensure a 444 chroma.

**Comment by Moarc** [comment]
So there's no way to use 4:4:4 RGB, like you can with heif-enc?

**Comment by urban-warrior** [comment]
Not without mods to the HEIC coder.  Refer to WriteHEICImage() in https://github.com/ImageMagick/ImageMagick/blob/main/coders/heic.c.  If you come up with a patch, post it here or as a pull request.  We'll look toward merging it into the main repo.
